File: week 6/Readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

letters = 0
words = 0
sentences = 0

text = text.lower()

counterLetter = len(
    text.replace(" ", "")
    .replace("!", "")
    .replace("?", "")
    .replace("'", "")
    .replace(".", "")
    .replace(",", "")
)
counterWords = len(text.split(" "))
counterSentences = len(text.replace("!", ".").replace("?", ".").split(".")) - 1

L = float(counterLetter) * 100 / float(counterWords)
S = float(counterSentences) * 100 / float(counterWords)
index = 0.0588 * L - 0.296 * S - 15.8


if index >= 16:
    print("Grade 16+")
elif index <= 1:
    print("Before Grade 1")
else:
    print(f"Grade {int(index + 0.5)}")

File: week 2/Readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{

    int letters = 0;
    int words = 0;
    int sentences = 0;

    string text = get_string("Text: ");
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }

        words = text[i] == ' ' ? words + 1 : words;

        sentences = (text[i] == '.' || text[i] == '!' || text[i] == '?') ? sentences + 1 : sentences;
    }
    words++;

    float L = (float) letters * 100 / (float) words;
    float S = (float) sentences * 100 / (float) words;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
        ;
    }
}
